### DISCUSSION TITLE
Attribution of revenue based on ROI * spend, dynamic nature

### DISCUSSION THREAD

**Discussion by christianlehre** [google/meridian#352] [Q&A] [answered]
Hello, 

I need some help with attributing revenue to our different marketing channels, in particular incorporating the dynamic nature of ROI into the attribution.

How can i correctly extract revenue from each channel using the estimated ROI-parameters? From posterior inference we get (by averaging over chains and draws), a single ROI-estimate for each channel - over the entire time period.

 If we naively multiply this estimate with the spend in each channel, we can attribute revenue into each channel, but this will only be a constant factor of the spend, which is false when we are looking over longer periods of time. How can we correctly incorporate a temporal dependency on the ROI of a channel? 

Do we need to run the media spend through HillAdstock before mutliplying with the constant ROI parameter estimate to get a proper estimate of the revenue attributed to spend in a channel?

**Comment by brendalewisprice** [comment, marked as answer]
It sounds like you are trying to calculate the expected impact of each media channel on revenue as modeled by your MMM over particular time periods. 

Meridian has a function in the analyzer class called[`incremental_impact`](https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer#incremental_impact) that will provide those estimates. By default it will provide estimates over the entire time range, however you can use the option `selected_times` to limit the estimates to particular time periods. 

If you need to calculate an ROI for a selected time period, you can use the function [`roi`](https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer#roi) in the analyzer class with the option 'selected_times' to get an ROI estimate for a time period of interest. Additionally, you can use the incremental impact estimates divided by spend for the same time period.
